projects/com.oracle.truffle.llvm.parser.factories/src/com/oracle/truffle/llvm/parser/factories/LLVMNativeIntrinsicsProvider.java

```diff
@@ -174,17 +180,110 @@ public LLVMNativeIntrinsicFactory(boolean forceInline, boolean forceSplit) {
         protected abstract RootCallTarget generate(FunctionType type);
     }
 
+    protected static class Demangler {
+        protected final List<UnaryOperator<String>> demanglerFunctions = Arrays.asList(this::demangleRustSymbol);
+
+        protected Optional<String> demangle(String name) {
```

Comment:
We never want demangler functions on the fastpath (in compiled code).
Please guard that with an CompilerAsserts.neverPartOfCompilation.